src/Icon.js

```diff
@@ -4,16 +4,22 @@ import { space, color } from 'styled-system'
 import PropTypes from 'prop-types'
 import icons from '../icons.json'
 
-// Support re-named icon names too
 // Should be removed eventually after v1.0.0
-const oldAndNewIcons = Object.assign({}, icons, {
+const aliases = {
   srollLeft: icons.chevronLeft,
   chevronLight: icons.chevronDown,
   chevronThick: icons.chevronDownThick
-})
+}
+
+const getPath = ({ name, next }) => {
+  if (next) {
```

Comment:
I have slight qualms about the nomenclature of the `next` prop, but the documentation does make it clear as to what purpose it serves.  Without documentation, something like `legacy={false}` or `legacySet={false}` would be easy to understand.  If you didn't want to provide a defaultProp for `legacy: true`, then something like `latestVersion` could also be concise.  Food for thought.

Reply:
Yeah, I always struggle with how to name these things. "New" would make sense, but it's obviously a reserved word... Open to suggestions for how to handle this. My other thought was to have a separate component for only the new icons

Reply:
I put an alternative API for switching between the legacy and new icons here: https://github.com/pricelinelabs/design-system/tree/new-icons-legacy-flag/

Instead of a prop on the Icon component, it uses the ThemeProvider's `legacy` prop, e.g.

```jsx
<ThemeProvider legacy>
  <Icon name='chevronDown' />
</ThemeProvider>
```

Would like to get others thoughts on how we should handle this

Reply:
I'll vote for `<Icon legacy />` with a deprecation warning for using legacy

Reply:
I like the second PR because I think it follows the same paradigm used for all the other properties in the theme. However, it could also be nice to be able to use a `new` icon if it doesn't exist on the `legacy` set and also force an icon to be `new` if the ThemeProvider is set to `legacy`. So, my take is maybe an alternative that mixes the two PRs.

Reply:
I should clarify that this *only* affects icons where the keys are the same from the legacy set to the new one - e.g. chevrons, calendar, and star

Reply:
I like the the legacy as prop on Icon

Reply:
Yeah, seems like the Icon prop is a more flexible solution. I'm going to change the name from `next` to `legacy` and get a non-breaking version out in a few minutes